dbuild/__init__.py

```diff
@@ -0,0 +1,146 @@
+    print '\n'.join(response)
+
+    if build_type == 'source':
+        command = ['dpkg-buildpackage', '-S', '-nc', '-uc', '-us']
+        cwd = '/build/' + source_dir
+    elif build_type == 'binary':
+        command = ['bash', '-c', "dpkg-source -x /build/*.dsc /build/pkgbuild/ && cd /build/pkgbuild && /usr/lib/pbuilder/pbuilder-satisfydepends && dpkg-buildpackage"]
+        cwd = '/build'
+    else:
+        raise Exception('Unknown build_type %s' % build_type)
+
+    container = create_container(c, image_tag,command=command, shared_volumes={build_dir: '/build'}, cwd=cwd)
+    print(container)
+    response = start_container(c, container)
+    rv = wait_container(c, container)
+    logs = container_logs(c, container)
+    print '\n'.join(logs)
+
+    if rv == 0:
+        print 'Build successful (build type: %s), removing container %s' % (build_type, container.get('Id'))
+        remove_container(c, container, force=True)
+        return True
+    else:
+        if force_rm:
+            print "Build failed (build type: %s), Removing container %s" % (build_type, container.get('Id'))
+            remove_container(c, container, force=True)
+            return False
+        else:
+            print "Build failed (build type: %s), keeping container %s" % (build_type, container.get('Id'))
+            return False
+
+def main(argv=sys.argv):
+    ap = argparse.ArgumentParser(description='Build debian packages in docker container')
+    ap.add_argument('--build_dir', type=str, help='package build directory')
+    ap.add_argument('--source_dir', type=str, default='source',
+            help='subdirectory of build_dir where sources kept')
+    ap.add_argument('--force_rm', action='store_true', default=False,
+            help='Remove the containers even if build failed')
+    ap.add_argument('--docker_url', type=str,
+            default='unix://var/run/docker.sock',
+            help='Docker url, it can be unix socket or tcp url')
+    ap.add_argument('--flavor', type=str, default='ubuntu',
+            help='Linux flavor to use for container to build')
+    ap.add_argument('--dist', type=str, default='trusty',
+            help='Linux distribution')
+
+    args = ap.parse_args()
+
+    PATH = os.path.dirname(os.path.abspath(__file__))
+
+    TMPL_ENV = Environment(
+            autoescape=False,
+            loader=FileSystemLoader(os.path.join(PATH, 'templates')),
+            trim_blocks=False)
+    docker_dir = mkdtemp()
+    dockerfile = os.path.join(docker_dir, 'Dockerfile')
+    ctxt = {'flavor': args.flavor, 'dist': args.dist,
+            'maintainer': 'dbuild, dbuild@test.com', }
+
+    with open(dockerfile, 'w') as d:
+        dockerdata = TMPL_ENV.get_template('dockerfile.jinja').render(ctxt)
+        d.write(dockerdata)
+
+    script_dir = os.path.join(PATH, 'scripts/')
+    for f in glob.glob(os.path.join(script_dir, '*')):
+        if os.path.isfile(f):
+            shutil.copy2(f, docker_dir)
```

Comment:
Ah I have seen this in the documentation, but I wanted to copy all files under a directory to separate existing directory, so used glob. But I can also change dockerfile such that it will look the scripts under a subdirectory, so we can use copytree.